plugins/azure/resoto_plugin_azure/resource/containerservice.py

```diff
@@ -0,0 +1,1139 @@
+#         "display": S("display") >> Bend(AzureOperationValueDisplay.mapping),
+#         "origin": S("origin"),
+#     }
+#     display: Optional[AzureOperationValueDisplay] = field(default=None, metadata={'description': 'Describes the properties of a Operation Value Display.'})  # fmt: skip
+#     origin: Optional[str] = field(default=None, metadata={"description": "The origin of the operation."})
+
+
+@define(eq=False, slots=False)
+class AzureKubernetesSnapshot(AzureResource):
+    kind: ClassVar[str] = "azure_kubernetes_snapshot"
+    api_spec: ClassVar[AzureApiSpec] = AzureApiSpec(
+        service="containerservice",
+        version="2023-08-01",
+        path="/subscriptions/{subscriptionId}/providers/Microsoft.ContainerService/snapshots",
+        path_parameters=["subscriptionId"],
+        query_parameters=["api-version"],
+        access_path="value",
+        expect_array=True,
+    )
+    mapping: ClassVar[Dict[str, Bender]] = AzureTrackedResource.mapping | {
+        "id": S("id"),
+        "tags": S("tags", default={}),
+        "name": S("name"),
+        "ctime": K(None),
+        "mtime": K(None),
+        "atime": K(None),
+        "creation_data_source_id": S("properties", "creationData", "sourceResourceId"),
+        "enable_fips": S("properties", "enableFIPS"),
+        "kubernetes_version": S("properties", "kubernetesVersion"),
+        "node_image_version": S("properties", "nodeImageVersion"),
+        "os_sku": S("properties", "osSku"),
+        "os_type": S("properties", "osType"),
+        "snapshot_type": S("properties", "snapshotType"),
+        "vm_size": S("properties", "vmSize"),
+    }
+    creation_data_source_id: Optional[str] = field(default=None, metadata={'description': 'Data used when creating a target resource from a source resource.'})  # fmt: skip
+    enable_fips: Optional[bool] = field(default=None, metadata={"description": "Whether to use a FIPS-enabled OS."})
+    kubernetes_version: Optional[str] = field(default=None, metadata={"description": "The version of Kubernetes."})
+    node_image_version: Optional[str] = field(default=None, metadata={"description": "The version of node image."})
+    os_sku: Optional[str] = field(default=None, metadata={'description': 'Specifies the OS SKU used by the agent pool. The default is Ubuntu if OSType is Linux. The default is Windows2019 when Kubernetes <= 1.24 or Windows2022 when Kubernetes >= 1.25 if OSType is Windows.'})  # fmt: skip
+    os_type: Optional[str] = field(default=None, metadata={'description': 'The operating system type. The default is Linux.'})  # fmt: skip
+    snapshot_type: Optional[str] = field(default=None, metadata={'description': 'The type of a snapshot. The default is NodePool.'})  # fmt: skip
+    vm_size: Optional[str] = field(default=None, metadata={"description": "The size of the VM."})
+
+
+resources: List[Type[AzureResource]] = [AzureManagedCluster, AzureFleet, AzureKubernetesSnapshot, AzureOSOptionProfile]
```

Comment:
I can see Fleet and ManagedCluster.
Let's discuss if we want to have AzureOSOptionProfile.
And what exactly are AzureKubernetesSnapshot resources?